test/python/algorithms/algorithms_test_case.py

```diff
@@ -12,10 +12,18 @@
 
 """Algorithms Test Case"""
 
+import warnings
 from qiskit.test import QiskitTestCase
 
 
 class QiskitAlgorithmsTestCase(QiskitTestCase):
     """Algorithms test Case"""
 
-    pass
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        allow_dep_warning_message = [
+            r"Implicit conversion from a PauliList to a SparsePauliOp*",
+        ]
+        for msg in allow_dep_warning_message:
+            warnings.filterwarnings("default", category=DeprecationWarning, message=msg)
```

Comment:
Since the algorithms will be removed, how about merging this PR after that?
https://github.com/Qiskit/qiskit/pull/11086

Reply:
Point 1 is still important because Algorithms isn't disappearing completely, it's just in a separate package. We should know how much work it is that we're asking downstream packages to do to manage a deprecation, and if the only option is a warning skip (though I don't think that is the case here), that's an indication that there are use cases that the new form cannot supply. It's a _good_ thing that this warning was triggered before Algorithms moved out and we'd have seen it too late.

The warnings and errors aren't there to be annoying, they're there to ensure that the effect of deprecations on consumers is properly accounted for, and skipping a warning that we emitted is very rarely the correct action.

Reply:
@chriseclectic I don't recall any algorithm/test in `qiskit.algorithms` explicitly using `PauliList`, but I guess that this warning filter was introduced after some tests raised warnings. Can you point out which tests did? This would help address Jake's first point.

Reply:
I expected that used around opflow, but not explicitly by algorithms. Searching here shows PauliList in 8 test files, one in opflow, nothing explicit in algorithms. Code use in primitives, visualization and of course quantum_info modules. Searching qiskit_algorithms repo shows no occurrence of PauliList at all

Reply:
I ran the test locally and found that no algorithm tests raise the deprecation warning due to PauliList. The deprecation warning is raised by only `test.python.primitives.test_estimator.TestObservableValidation.test_validate_observables_deprecated_{1,2}` as expected.

Reply:
Right. I ran the test locally and all tests passed.